### DISCUSSION TITLE
Is there a way to transcribe an english audio file to text then have the text language changed into different one?

### DISCUSSION THREAD

**Discussion by khmanaseer** [openai/whisper#1539] [Show and tell]
Is there a way to transcribe an English audio file to text then have the text language changed into different one?, if its not possible, whats the most reliable or accurate way to translate .srt file without doing it manually

Appreciate your help.

**Comment by misutoneko** [comment]
Yes. I've used [opusMT](https://github.com/Helsinki-NLP/OPUS-MT-app) for this purpose, but there are other options.
My translations were mostly english to finnish, with varied results. Not too bad, but not perfect either.
But ofc the results will heavily depend on what your actual target language is.

Note that if you need srt, you will need some way to deal with the timestamps.
(They probably won't survive going through the translation process).

**Reply by khmanaseer** [reply]
Unfortunately I need it .srt, I'm currently using Chat GPT for translation, no matter what prompt I use, it will sometimes blow the timestamp along with my head, I ll give opusMT a try, thanks ::)

**Reply by misutoneko** [reply]
Right, in that case you will need something that handles the timestamps separately (opusMT doesn't know how to handle them).
I've written some wrapper scripts that rip out the lines with timestamps and put them back after the translation, and that approach sorta works.
Actually I was thinking about using a local LLM for this, but I guess if ChatGPT can't handle it, my tiny 7B LLMs have no chance :(

There are some other challenges too, like sentences split into multiple parts, and real names (they get translated along with everything else).

**Reply by khmanaseer** [reply]
I think Bard can do the job for you, it doesn't work for me because Im doing a language that reads from right to left,  therefor my timestamp changes as well and become unreadable, otherwise its great. Chat GPT understood the task and kind of did good job on some files, while some others are doomed to fail.

**Reply by misutoneko** [reply]
Yes that's a good point, ChatGPT isn't everything. There's a lot of variation out there.
I'm obsessed in keeping everything offline, so that's one reason why I found opusMT a good fit for me.
(Other being that it actually has a model for finnish, tiny as that might be).

On those failed files, did you ever find out what the problem with ChatGPT was?
I realize that with LLMs it might not be very straightforward to find out...
But I wonder if it worked better if you gave it a smaller chunk to process, or prompted it somehow (to check its results, for example).

**Comment by phineas-pta** [comment]
https://github.com/baxtree/subaligner

**Reply by khmanaseer** [reply]
Thanks alot
